### PR TITLE
Don't complete statement when in multi-line verbatim string that is passed to a method

### LINKED ISSUE #34176: Complete Statement behaves incorrectly in a multi-line verbatim string that is passed to a method
adding notes from taylor 
image

_Originally posted by @jinujoseph in https://github.com/dotnet/roslyn/issues/34088#issuecomment-473122101_

### PR DESCRIPTION
**Pull request by chborl**
Fixes #34176

### CODE REVIEW

**Comment by chborl** [comment]
@dotnet/roslyn-ide  for review

File: src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs [hunk 1]
```diff
@@ -242,9 +242,9 @@ private static bool IsStatementOrFieldDeclaration(SyntaxNode currentNode, ISynta
             => syntaxFacts.IsStatement(currentNode) || currentNode.IsKind(SyntaxKind.FieldDeclaration);
 
         private static bool IsInAString(SyntaxNode currentNode, SnapshotPoint caret)
-            // If caret is at the end of the line, it is outside the string	
+            // If caret is at the end of the string, and at the end of the line, it's outside the string
             => currentNode.IsKind(SyntaxKind.InterpolatedStringExpression, SyntaxKind.StringLiteralExpression)
-                && caret.Position != caret.GetContainingLine().End;
+                && !(caret.Position == currentNode.Span.End && caret.Position == caret.GetContainingLine().End);
```
Comment (sharwell) [inline src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs]: :question: Why can't this just be:

```csharp
&& caret.Position < currentNode.Span.End
```

Comment (chborl) [reply src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs]: Ah yes, that is a much better way to check.  :)

**Comment by chborl** [comment]
@sharwell , I added the tests and a check for the caret before the string.

**Pull request merged**